frontend/src/globals/unity-indexedbcache.ts

```diff
@@ -0,0 +1,177 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * An IndexedDbCache instance acts as the counterpart of an IndexedDbCache
+ * Component in Unity. When the Component is started in Unity, an instance of
+ * this class is created by UnityUtil and attached to the unityInstance.
+ * This reference is used by functions in the indexedDb.jslib plug-in in Unity
+ * to make calls directly from the viewer. This class receives requests to
+ * create transactions against IndexedDb, and returns their results through
+ * SendMessage. In practice, IndexedDb operations are offloaded to a WebWorker,
+ * so this implementation is primarily used to maintain the worker and translate
+ * between it and Unity.
+ */
+export class IndexedDbCache {
+	constructor(unityInstance: any, gameObjectName: string) {
+		this.unityInstance = unityInstance;
+		this.gameObjectName = gameObjectName;
+		this.transactions = {};
+		this.createWorker();
+	}
+
+	/** The Unity Instance created by the loader; this is used for SendMessage */
+	unityInstance: any;
+
+	/** The gameObject which should be passed to SendMessage when making the
+	 * callbacks to the viewer */
+	gameObjectName: string;
+
+	/** Resources for in-progress get requests */
+	transactions: any;
+
+	/** The WebWorker that handles the IndexedDb requests. Each instance of this
+	 * class gets its own worker. */
+	worker: any;
+
+	// IndexedDb actions are offloaded to a WebWorker. This is for performance
+	// reasons. WebWorkers run individual scripts in their own thread, and
+	// interact with the main thread (of their creator) by posting messages.
+	// When messages are passed, a deep copy is created.
+
+	createWorker() {
+		this.worker = new Worker(new URL('dist/indexeddbworker.js', window.location.origin));
```

Comment:
could we put this in the unity folder instead of dist?

Reply:
Updated! I added a new member to UnityUtil (I also finally understand WebPack enough now to move the folder!)